examples/leaderregistry/main_test.go

```diff
@@ -498,6 +498,58 @@ func TestSurviveReplicaFailureWithRandomStrategy(t *testing.T) {
 	}
 }
 
+// TestRemoteServerTimeout tests the scenario where an actor is invoked with 2 replicas.
+// The invocation is done using a context with a deadline, and invoking a special operation that receives the expected
+// deadline as the payload. The actor operation checks whether the expected deadline matches the context deadline received by
+// the server. Due to HTTP lag, the test verifies that the deadlines are no more than 1 second apart.
+func TestRemoteServerTimeout(t *testing.T) {
+	t.Parallel()
+
+	var (
+		lp          = &leaderProvider{}
+		portServer1 = nextPort()
+	)
+
+	// Set the leader address for the registry.
+	lp.setLeader(registry.Address{
+		IP:   net.ParseIP("127.0.0.1"),
+		Port: baseRegistryPort + portServer1,
+	})
+
+	var (
+		server1, _, cleanupFn1 = newServer(t, lp, portServer1)
+		server2, _, cleanupFn2 = newServer(t, lp, nextPort())
+	)
+
+	// Clean up resources at the end of the test.
+	defer cleanupFn1()
+	defer cleanupFn2()
+	defer server1.Close(context.Background())
+	defer server2.Close(context.Background())
+
+	// Sleep for a few seconds to allow servers to heartbeat and actors to activate.
+	time.Sleep(5 * time.Second)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Define the actor options with 1 extra replica and the random selection strategy.
+	options := types.CreateIfNotExist{Options: types.ActorOptions{
+		ExtraReplicas:       1,
+		ReplicationStrategy: types.ReplicaSelectionStrategyRandom,
+	}}
+
+	// Use require.Eventually to invoke the actor until it is replicated in all available servers.
```

Comment:
Its not really clear why this is important. I assume you want to guarantee it reached two different servers to ensure at least one of your requests went over the network right? Can you comment that? Also make sure that ForceRemoteProcedureCalls is enabled and call that out in the comment as well